src/UserService.Data/UserCredentialsRepository.cs

```diff
@@ -105,12 +105,31 @@ public bool Edit(DbUserCredentials userCredentials)
 
         public Guid Create(DbUserCredentials dbUserCredentials)
         {
+            if (dbUserCredentials == null)
+            {
+                throw new ArgumentNullException(nameof(dbUserCredentials));
+            }
+
             _provider.UserCredentials.Add(dbUserCredentials);
             _provider.Save();
 
             return dbUserCredentials.Id;
         }
 
+        public bool Remove(Guid userId)
+        {
+            DbUserCredentials credentials = _provider.UserCredentials.FirstOrDefault(c => c.UserId == userId);
+
+            if (credentials != null)
+            {
+                _provider.UserCredentials.Remove(credentials);
```

Comment:
we talk about deactivated, not removing

Reply:
We deactivated user, credentials we can remove